crates/bevy_asset/src/io/mod.rs

```diff
@@ -27,20 +27,34 @@ use futures_lite::{ready, Stream};
 use std::{
     path::{Path, PathBuf},
     pin::Pin,
+    sync::Arc,
     task::Poll,
 };
 use thiserror::Error;
 
+use crate::{retry::AssetLoadRetrySettings, UntypedAssetLoadFailedEvent};
+
 /// Errors that occur while loading assets.
-#[derive(Error, Debug)]
+#[derive(Error, Debug, Clone)]
 pub enum AssetReaderError {
     /// Path not found.
-    #[error("path not found: {0}")]
+    #[error("Path not found: {0}")]
     NotFound(PathBuf),
 
     /// Encountered an I/O error while loading an asset.
-    #[error("encountered an io error while loading asset: {0}")]
-    Io(#[from] std::io::Error),
+    #[error("Encountered an I/O error while loading asset: {0}")]
+    Io(Arc<std::io::Error>),
```

Comment:
Why do we need an Arc here?

Reply:
It's needed because `AssetReaderError` is now cloned and passed around places and `std::io::Error` does not implement `Clone`.